src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -123,6 +124,65 @@ namespace xclhwemhal2 {
     return buf_size;
   }
 
+  void HwEmShim::writeNewSimulateScript(std::string simPath, std::string simName, std::string simCmd, std::string guiOrBatch, std::string guiEnableOption)
+  {
+
+    std::stringstream ss;
+    ss << simPath << "/simulate.sh";
+    std::string filePth = ss.str();
+
+    std::ifstream seek1;
+    std::ofstream temp1;
+    seek1.open(filePth);
+    temp1.open(simPath + "/temp.sh");
+    std::string temp;
```

Comment:
Ouch... Perhaps you could spend some time studying https://en.cppreference.com/w/ :-)
```c++
std::ifstream seek1 { simPath + "/simulate.sh" };
```
and so on.